agent/engine/docker_client.go

```diff
@@ -588,32 +582,28 @@ func dockerStateToState(state docker.State) api.ContainerStatus {
 	return api.ContainerStopped
 }
 
-func (dg *dockerGoClient) DescribeContainer(dockerID string) (api.ContainerStatus, DockerContainerMetadata) {
-	dockerContainer, err := dg.InspectContainer(dockerID, inspectContainerTimeout)
+func (dg *dockerGoClient) DescribeContainer(ctx context.Context, dockerID string) (api.ContainerStatus, DockerContainerMetadata) {
+	dockerContainer, err := dg.InspectContainer(ctx, dockerID, inspectContainerTimeout)
 	if err != nil {
 		return api.ContainerStatusNone, DockerContainerMetadata{Error: CannotDescribeContainerError{err}}
 	}
 	return dockerStateToState(dockerContainer.State), metadataFromContainer(dockerContainer)
 }
 
-func (dg *dockerGoClient) InspectContainer(dockerID string, timeout time.Duration) (*docker.Container, error) {
+// InspectContainer uses the timeout parameter only to assemble the return error message
+// TODO: replace the timeout with a string type error message
```

Comment:
Can you get rid of this TODO (and others that were added)? It's out of date now.

Reply:
This is the one that I added, we have talked about changing the timeout parameter into some kind of error message, and add a TODO for this purpose.

Reply:
Right, that was when timeout was only being used for generating error messages. But, now its actually used for creating the context. So, that's not needed anymore

Reply:
Oh I see, you are right!